rcl/include/rcl/node.h

```diff
@@ -0,0 +1,228 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__NODE_H_
+#define RCL__NODE_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/allocator.h"
+#include "rcl/types.h"
+
+// Intentional underflow to get max size_t.
+#define RCL_NODE_OPTIONS_DEFAULT_DOMAIN_ID (size_t)-1
+
+struct rcl_node_impl_t;
+
+/// Handle for a ROS node.
+/* This handle can be in one a few stats, which are referred to below:
+ * - uninitialized: memory allocated, but not initialized to anything
+ * - invalid: memory allocated but not usable
+ *   - either zero initialized or shutdown (rcl_fini() or rcl_node_fini())
+ * - valid: memory allocated and usable
+ */
+typedef struct rcl_node_t {
+  /// Private implementation pointer.
+  struct rcl_node_impl_t * impl;
+} rcl_node_t;
+
+typedef struct rcl_node_options_t {
+  // bool anonymous_name;
+  // rmw_qos_profile_t parameter_qos;
+  /// If true, no parameter infrastructure will be setup.
+  bool no_parameters;
+  /// If set, then this value overrides the ROS_DOMAIN_ID environment variable.
+  /* It defaults to RCL_NODE_OPTIONS_DEFAULT_DOMAIN_ID, which will cause the
+   * node to use the ROS domain ID set in the ROS_DOMAIN_ID environment
+   * variable, or on some systems 0 if the environment variable is not set.
+   *
+   * \TODO(wjwwood): Should we put a limit on the ROS_DOMAIN_ID value, that way
```

Comment:
This is a good question. I don't know what the limits are for the DDS vendors or even if the specification deals with that detail. This remains an open question for me.